src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs

```diff
@@ -491,7 +491,7 @@ public class DependencyGraph
             HashSet<DependencyGraphNode> incoherentNodes = new HashSet<DependencyGraphNode>();
             // Cache of incoherent dependencies, looked up by name
             Dictionary<string, DependencyDetail> incoherentDependenciesCache = new Dictionary<string, DependencyDetail>();
-            HashSet<DependencyDetail> incoherentDependencies = new HashSet<DependencyDetail>();
+            HashSet<DependencyDetail> incoherentDependencies = new HashSet<DependencyDetail>(new DependencyDetailComparer());
```

Comment:
I think you actually want to use a  slight variant of the DPC here:

Currently it does:

            return x.Commit == y.Commit &&
                x.Name == y.Name &&
                x.RepoUri == y.RepoUri &&
                x.Version == y.Version &&
                x.Type == y.Type;

Type does not matter in the case we are talking about, and I think you can make an argument that repo doesn't either.